Handle the new package structure

ekohl opened the pull request:
Since RHEL 6.10 and RHEL 7.5 the package was renamed from redhat-access-insights to insights-client. When systemd is present, a timer unit is used instead of cron.

We use facts to automatically determine which package should be present, but this can fail when content views are used. Because of that a new parameter is introduced.

**Note** Please let me know if this is correct. I haven't had a chance to test this on a real system yet.

ekohl committed:
Update summary

ekohl committed:
Handle the new package structure

Since RHEL 6.10 and RHEL 7.5 the package was renamed from
redhat-access-insights to insights-client. When systemd is present, a
timer unit is used instead of cron.

We use facts to automatically determine which package should be present,
but this can fail when content views are used. Because of that a new
parameter is introduced.

ehelms commented:
@lphiri Can you take a look at this?

lphiri reviewed (changes requested):
@ekohl 
Apparently the weekly schedule option is no supported so we need to remove it.
Otherwise changes look ok to me.

vjsaisha commented:
@lphiri when we can expect this module in puppet forge website?

ekohl commented:
@lphiri did you want me to remove the week schedule as part of this PR?

lphiri commented:
@ekohl yes please.
@rexwhite

ekohl committed:
Remove upload_schedule parameter and always use daily

The weekly upload schedule is no longer supported and this ensures we're
on a daily schedule.

ekohl commented:
Added a commit that removes the parameter. This does make it a breaking change and you must be sure the plugin isn't sending the parameter. It does remove a TODO so that's always a nice thing :)

ekohl commented:
From internal testing:
```
# ls -ltr /etc/insights-client/insights-client.cron
ls: cannot access /etc/insights-client/insights-client.cron: No such file or directory
```

I'll work on an update

treydock commented:
I am testing this PR with Satellite 6.4 and noticed the INI section for `/etc/insights-client/insights-client.conf` is actually `insights-client` from stock install on RHEL 7.6:

```
--- /etc/insights-client/insights-client.conf   2018-01-30 14:08:27.000000000 -0500
+++ /tmp/puppet-file20190228-53440-gvu4ve       2019-02-28 16:47:40.320439000 -0500
@@ -1,39 +1,20 @@
-[insights-client]
-# Example options in this file are the defaults
+[redhat-access-insights]
+
+
+auto_config=True
+
+
+
+
 
-# Change log level, valid options DEBUG, INFO, WARNING, ERROR, CRITICAL. Default DEBUG
-#loglevel=DEBUG
 
-# Log each line executed
-#trace=False
 
-# Attempt to auto configure with Satellite server
-#auto_config=True
 
-# Change authentication method, valid options BASIC, CERT. Default BASIC
-#authmethod=BASIC
 
-# username to use when authmethod is BASIC
-#username=
-# password to use when authmethod is BASIC
-#password=
 
-#base_url=cert-api.access.redhat.com:443/r/insights
 
-# URL for your proxy.  Example: http://user:pass@192.168.100.50:8080
-#proxy=
 
-# Automatically update the dynamic configuration
-#auto_update=True
 
-# Obfuscate IP addresses
-#obfuscate=False
 
-# Obfuscate hostname
-#obfuscate_hostname=False
 
-# Disallow Insights from creating cron job
-#no_schedule=False
 
-# Display name for registration
-#display_name=
```

treydock commented:
I think this should fix incorrect section name

```diff
--- a/templates/redhat-access-insights.conf.erb
+++ b/templates/redhat-access-insights.conf.erb
@@ -1,4 +1,4 @@
-[redhat-access-insights]
+[<%= @package_name %>]
 
 <%- if @log_level -%>
 loglevel=<%= @log_level %>
```

treydock commented:
Also need this:

```diff
--- a/manifests/current.pp
+++ b/manifests/current.pp
@@ -77,8 +77,9 @@ class access_insights_client::current (
     }
 
     service { "${package_name}.timer":
-      ensure => 'running',
-      enable => true,
+      ensure  => 'running',
+      enable  => true,
+      require => Package[$package_name],
     }
   } else {
     file { "/etc/cron.daily/${package_name}":
```

lphiri commented:
Closing this PR to continue on #5
